translator/util/ecsutil/ecsutil.go

```diff
@@ -65,26 +68,29 @@ func (e *ecsUtil) IsECS() bool {
 }
 
 func (e *ecsUtil) getECSMetadata() (em *ecsMetadataResponse, err error) {
-	// choose available endpoint
-	if v3MetadataEndpoint, ok := os.LookupEnv(v3MetadataEndpointEnv); !ok {
-		em, err = e.getMetadataResponse(v2MetadataEndpoint)
-	} else {
+	// Based on endpoint to get ECS metadata, for more information on the respond, https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-metadata-endpoint.html
+	if v4MetadataEndpoint, ok := os.LookupEnv(v4MetadataEndpointEnv); ok {
+		em, err = e.getMetadataResponse(v4MetadataEndpoint + "/task")
+	} else if v3MetadataEndpoint, ok := os.LookupEnv(v3MetadataEndpointEnv); ok {
 		em, err = e.getMetadataResponse(v3MetadataEndpoint + "/task")
+	} else {
+		em, err = e.getMetadataResponse(v2MetadataEndpoint)
 	}
 	return
 }
 
 func (e *ecsUtil) getMetadataResponse(endpoint string) (em *ecsMetadataResponse, err error) {
 	em = &ecsMetadataResponse{}
 	resp, err := e.httpClient.Request(endpoint)
+	
 	if err != nil {
 		return
 	}
 
 	err = json.Unmarshal(resp, em)
 	if err != nil {
-		log.Printf("E! unable to parse resp from ecsmetadata endpoint, error: %v", err)
-		log.Printf("D! resp content is %s", string(resp))
+		log.Printf("E! Unable to parse response from ecsmetadata endpoint, error: %v", err)
+		log.Printf("D! Respond content is %s", string(resp))
```

Comment:
Nitpick: I think this should be `Response content`, not `Respond content`. Or just drop it entirely:
```suggestion
		log.Printf("D! Content is %s", string(resp))
```